pyfr/partitioners/base.py

```diff
@@ -422,7 +422,8 @@ def partition(self, mesh, progress=NullProgressSequence):
                     emap[k] = eleglmap[etype, eidx]
 
         # Generate a new UUID for the mesh
-        newmesh['mesh_uuid'] = newuuid = str(UUID(digest(curruuid, rnum)[:32]))
+        newuuid = np.array(str(UUID(digest(mesh)[:32])), dtype='S')
```

Comment:
Not sure this change is needed?

Reply:
I thought that it was best if, throughout the code, the UUID was written in the same way, even if we support the edge case of it just being a string.

Reply:
Is encode() needed?  As we have it in the file writer when writing strings.

Reply:
Just looking, the str() call is not needed.